### PR TITLE
Add release procedure documentation

### PR DESCRIPTION
**Pull request by copejon**
Publish CDI versioning and release procedures for developer reference

Addresses #205

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,16 @@ This approach supports two main use-cases:
 
 For an in depth look at the system and workflow, see the [Design](/doc/design.md#design) documentation.
 
+### Versions
```
Comment (screeley44) [inline README.md]: I don't see a link to the releases.md, we probably want that right?

**Comment by screeley44** [comment]
other than minor nit above, lgtm

File: README.md (hunk 1 above)
Comment (jeffvance) [inline README.md]: What a version link in the TOC?

File: README.md [hunk 2]
```diff
@@ -21,6 +21,16 @@ This approach supports two main use-cases:
...
+
+CDI follows the common semantic version scheme defined at semver.org in the "vMajor.Minor.Patch" pattern.  These are defined as:
```
Comment (jeffvance) [inline README.md]: add link in this paragraph to to the release doc

File: README.md (hunk 1 above)
Comment (copejon) [reply README.md]: Yeah, good point, will add

Comment (copejon) [reply README.md]: Will do

File: doc/releases.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+- Minor - End of Sprint release. Encapsulates non-API-breaking changes within the current Major version.  The current Sprint cycle is 2 weeks long, producing in bug fixes and feature additions.  Publishing a Minor version at the end of the cycle allows consumers to immediately access the end product of that Sprint's goals. Issues or bugs can be reported and addressed in the following Sprint.  It is expected that this patch contain myriad commits.
+
+- Patch - mid-Sprint release for fixing blocker bugs. In the case that a bug is blocking CDI consumers' workflow, a fix may be released as soon as it is merged.  A Patch should be limited expressly to the bug fix and not include anything unrelated.
+
+### Releasing a New Version
+
+ The version number is tracked in several files in CDI as well as through a git tag.  To reduce the chance of human error, a help script is used to change the version in all known locations.
+
+     DO NOT EDIT THE VERSION VALUES!!
```
Comment (jeffvance) [inline doc/releases.md]: How about **DO NOT EDIT ANY VERSION STRINGS IN CDI!**?

**Pull request merged**